Content.Client/ShortConstruction/UI/ShortConstructionMenuBUI.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System.Numerics;
+using Content.Client.Construction;
+using Content.Client.UserInterface.Controls;
+using Content.Shared.Construction.Prototypes;
+using Content.Shared.ShortConstruction;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Input;
+using Robust.Client.Placement;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Shared.Enums;
+using Robust.Shared.Prototypes;
+
+// ReSharper disable InconsistentNaming
+
+namespace Content.Client.ShortConstruction.UI;
+
+[UsedImplicitly]
+public sealed class ShortConstructionMenuBUI : BoundUserInterface
+{
+    [Dependency] private readonly IClyde _displayManager = default!;
+    [Dependency] private readonly IInputManager _inputManager = default!;
+    [Dependency] private readonly EntityManager _entManager = default!;
+    [Dependency] private readonly IPrototypeManager _protoManager = default!;
+    [Dependency] private readonly IPlacementManager _placementManager = default!;
+
+    private readonly ConstructionSystem _construction;
+    private readonly SpriteSystem _spriteSystem;
+
+    private RadialMenu? _menu;
+
+    public ShortConstructionMenuBUI(EntityUid owner, Enum uiKey) : base(owner, uiKey)
+    {
+        _construction = _entManager.System<ConstructionSystem>();
+        _spriteSystem = _entManager.System<SpriteSystem>();
+    }
+
+    protected override void Open()
+    {
+        _menu = FormMenu();
+        _menu.OnClose += Close;
+        _menu.OpenCenteredAt(_inputManager.MouseScreenPosition.Position / _displayManager.ScreenSize);
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        base.Dispose(disposing);
+        if (disposing)
+            _menu?.Dispose();
+    }
+
+    private RadialMenu FormMenu()
+    {
+        var menu = new RadialMenu
+        {
+            HorizontalExpand = true,
+            VerticalExpand = true,
+            BackButtonStyleClass="RadialMenuBackButton",
+            CloseButtonStyleClass="RadialMenuCloseButton"
+        };
+
+        var mainContainer = new RadialContainer
+        {
+            Radius = 100
```

Comment:
Do not hardcode the radius, or the menu will break like emoji wheel on wizden. Instead, calculate it based on the number of elements. Assuming the menu is a hexagon with n sides and side length 48, its radius should be `48/2 / sin(PI / n / 2)` or `24 / sin(PI / n / 2)`; in this case it would be

```cs
            Radius = 24f / Math.Sin(Math.PI / (crafting.Prototypes.Count - 1) / 2)
```

Though you can set it the base factor higher than 24 to add padding.

Reply:
Thanks! I was really wondering about the formula for it.

Reply:
![image](https://github.com/user-attachments/assets/10e70be5-3c3a-4488-9f54-1873a5f26400)
Great

Reply:
Uh, little note... I didn't consider that `crafting.Prototypes.Count - 1` may result in division by 0, and that there's a button in the middle... I guess the `- 1` part should be removed since both are the case.